src/convertToObject.js

```diff
@@ -6,7 +6,25 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const source = sourceString;
+  const splitedByASemicolon = source
+    .split(';')
+    .map((string) => string.trim())
+    .filter((string) => string.length > 0);
+  const arrOfArrsSplitedByAColon = [];
+
+  splitedByASemicolon.forEach((string) => {
+    arrOfArrsSplitedByAColon.push(string.split(':'));
+  });
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. The `forEach` loop on lines 16-18 could be replaced with a `map` function that returns the split arrays directly.

Reply:
Is it really necessary?